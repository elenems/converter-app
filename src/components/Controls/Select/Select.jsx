import React from "react";
import MaterialSelect from "@material-ui/core/Select";
import MenuItem from "@material-ui/core/MenuItem";
import FormControl from "@material-ui/core/FormControl";
import InputLabel from "@material-ui/core/InputLabel";
import OutlinedInput from "@material-ui/core/OutlinedInput";

export default React.memo(function Select({
  selectId,
  selectLabel,
  values,
  selectValue,
  handleChange
}) {
  const inputLabel = React.useRef(null);

  const [labelWidth, setLabelWidth] = React.useState(0);
  React.useEffect(() => {
    setLabelWidth(inputLabel.current.offsetWidth);
  }, []);

  return (
    <FormControl variant="outlined">
      <InputLabel ref={inputLabel} htmlFor={selectId}>
        {selectLabel}
      </InputLabel>
      <MaterialSelect
        onChange={handleChange}
        value={selectValue}
        input={
          <OutlinedInput
            labelWidth={labelWidth}
            name={selectId}
            id={selectId}
          />
        }
      >
        {values.map(value => {
          return (
            <MenuItem key={value} value={value}>
              {value}
            </MenuItem>
          );
        })}
      </MaterialSelect>
    </FormControl>
  );
});
