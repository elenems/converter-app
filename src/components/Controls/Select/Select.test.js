import React from 'react';
import { shallow } from 'enzyme';
import {currencies} from '../../../data/currencies';
import Select from './Select';

describe('Select', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<Select values = {currencies} />).debug();
    expect(component).toMatchSnapshot();
  });
});