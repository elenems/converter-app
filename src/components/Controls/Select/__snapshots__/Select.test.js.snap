// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;
